### ISSUE TITLE
tag 1.0.0 was deleted without notice which caused errors with glide

### ISSUE THREAD

**Issue by krnkl** [justinas/alice#42]
Hi, we've just noticed that our build jobs started failing because of the deleted tag in your repository.

we use glide with a following configuration:

> - package: github.com/justinas/alice
>   version: ^1.0.0 

However, with this configuration glide fails to fetch corresponding alice package
To reproduce issue locally, first remove glide cache under `~/.glide/`

glide version:
```
$ glide -v
glide version v0.12.3
```

**Comment by justinas** [comment]
Hi @krnkl,

Sorry for breaking your build :disappointed: After [this issue](https://github.com/justinas/nosurf/issues/44) was raised in another project of mine, I decided to get rid of tags in both nosurf in alice, as I never got around to actually versioning the projects meaningfully.

Your best bet would be to pin to the current master: from what I remember 1.0.0 was pretty old and no newer tags were ever created. I am going to think about embracing semver, but for now alice will probably follow the classic Go way of not having any meaningful versions.

Sorry again and hopefully this helps.

**Comment by SeerUK** [comment]
I'm hoping that you'll reconsider this, given that tools like Glide and [dep](https://github.com/golang/dep) are only more likely to pick up traction. SemVer is really useful, and can help avoid a lot of headaches. The reason dep is being made is because the Go team has acknowledged those issues.

**Comment by justinas** [comment]
I will certainly consider this. But as of today I still suggest pinning to 1051eaf52. Without a real versioning process on my side, it is no more or less `1.0` or `1.1` or `2.0` than the commit that the stale `1.0.0` tag was pointing to.

**Comment by justinas** [comment]
To add to that: I realize the mistake that just deleting the tag outright was, but when tags are not maintained, keeping an outdated tag as "the latest stable version" is still less than ideal as it prevents people from getting the latest improvements (even with a perfect API stability).

**Comment by SeerUK** [comment]
The decision is understandable, absolutely. There's not much else to say I suppose. Thank you for continuing to maintain the library - and I hope that versioning makes a comeback in the near future. I'd recommend against making another v1.0.0 though and maybe going for v2.0.0 straight off the bat.

**Comment by scottcrespo** [comment]
I just experienced the same problem with the deleted tag while using glide. I definitely recommend using semantic versioning. Since Go's dependency management is so tightly integrated with source control, it's pretty vital, in my opinion, to maintain thorough semantic versioning. Even for our closed-source, we semver all the things!

Still, the alice library is great, and I've been using it in various projects for years now. And, I very much appreciate your contributions. As a workaround, we've forked the project to maintain a set of tags, and we'll pull periodically to stay up to date with any changes in the parent project.  

Cheers!
